merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -51,11 +51,18 @@ public List<ConstraintResult> getViolations(final PlanId planId, final Optional<
       throws NoSuchPlanException, MissionModelService.NoSuchMissionModelException, SimulationDatasetMismatchException {
     final var plan = this.planService.getPlanForValidation(planId);
     final Optional<SimulationResultsHandle> resultsHandle$;
+    final SimulationDatasetId simDatasetId;
     if (simulationDatasetId.isPresent()) {
       resultsHandle$ = this.simulationService.get(planId, simulationDatasetId.get());
+      simDatasetId = resultsHandle$
+        .map(SimulationResultsHandle::getSimulationDatasetId)
+        .orElseThrow(() -> new InputMismatchException("simulation dataset with id `" + simulationDatasetId.get().id() + "` does not exist"));
     } else {
       final var revisionData = this.planService.getPlanRevisionData(planId);
       resultsHandle$ = this.simulationService.get(planId, revisionData);
```

Comment:
My expectation is that `.get` returns `Optional.empty()` in the event that this plan has been simulated before, but it has been edited since the last simulation, so there are no up-to-date simulation datasets. Does that track?

Reply:
I thought it didn't based on the error message, but some testing indicates that it does. I have adjusted the message 👍.